### PR TITLE
cpu/driver: handle older intel systems which do not have pstate support

### PR DESCRIPTION
**Pull request by darcari**
On some older intel systems the intel_pstate driver is not supported.
We expect those systems to be running acpi-cpufreq.  Avoid returning
FAIL in those cases.

### CODE REVIEW

**Comment by darcari** [comment]
v3: made changes suggested in v2

Test results here:

https://beaker.engineering.redhat.com/jobs/3852402
https://beaker.engineering.redhat.com/jobs/3852401

File: cpu/driver/runtest.sh [hunk 1]
```diff
@@ -25,6 +25,25 @@ TMPDIR=/var/tmp/$(date +"%Y%m%d%H%M%S")
 
 source $CDIR/../../cpu/common/libutil.sh
 
+NON_PSTATE_PROCESSORS="26 31 46"
+check_pstate_support()
+{
+    typeset model=$(lscpu | grep Model: | awk '{print $2}')
+    if [ -z "$model" ]; then
+	cki_log "unable to determine cpu model"
+	return 1
```
Comment (rasibley) [inline cpu/driver/runtest.sh]: @darcari would you consider this a kernel bug or a setup issue, just wanted to confirm it should be a fail in this case ?

File: cpu/driver/runtest.sh [hunk 2]
```diff
@@ -25,6 +25,25 @@ TMPDIR=/var/tmp/$(date +"%Y%m%d%H%M%S")
...
+    fi
+
+    for m in $NON_PSTATE_PROCESSORS; do
+	if [ $m -eq $model ]; then
+	    cki_log "model: $model - does not support intel_pstate"
+	    return 1
```
Comment (rasibley) [inline cpu/driver/runtest.sh]: This sounds like it's unsupported, should it return a skip ?

File: cpu/driver/runtest.sh [hunk 3]
```diff
@@ -34,6 +53,18 @@ function verify_intel_cpufreq_driver
     typeset vendor=$(dmidecode -t 0 | grep Vendor: | \
                      cut -d: -f 2 | awk '{print tolower($1)}')
 
+    # make sure the given system supports p-state
+    check_pstate_support
+    if [ $? -ne 0 ]; then
+	# older systems do not support intel pstate
+	if [ $driver != "acpi-cpufreq" ]; then
+	    cki_log "intel (non-pstate) system is running: $driver"
+	    return $CKI_FAIL
```
Comment (rasibley) [inline cpu/driver/runtest.sh]: should we skip instead ?

File: cpu/driver/runtest.sh (hunk 1 above)
Comment (darcari) [reply cpu/driver/runtest.sh]: I don't really think it is possible.

File: cpu/driver/runtest.sh (hunk 3 above)
Comment (darcari) [reply cpu/driver/runtest.sh]: I we'd typically expect some cpu frequency driver to be running.  I guess they are all old systems so perhaps SKIP is better.

File: cpu/driver/runtest.sh (hunk 2 above)
Comment (darcari) [reply cpu/driver/runtest.sh]: This is in check_pstate_support() which returns a 0 on success (meaning pstate is supported) and 1 on failure (meaning it is not).  So this is correct.

**Pull request merged**